### PR TITLE
Filled in typehints and doctrings for fast64_internal/utility.py

### PR DESCRIPTION
**Pull request by ThomasJRyan**
What's the best way to jump into a project? Read the documentation!

No documentation? Second best way then, write the documentation!

Docstrings are heavily written by Copilot, typehints are all traced back by myself. I'm reasonably certain of the accuracy of it all.

### CODE REVIEW

**Comment by sauraen** [comment]
Thanks for the submission!

There are already some conflicts, but there is another PR or two going in which touch this file too. Once that happens please merge main to your branch and fix the conflicts, then update this branch. Feel free to tag me to review then.

File: fast64_internal/utility.py [hunk 1]
```diff
@@ -137,33 +183,56 @@ def restoreHiddenState(hiddenState):
         layerCol.hide_viewport = True
 
 
-def readFile(filepath):
+def readFile(filepath: str | Path) -> str:
```
Comment (Lilaa3) [inline fast64_internal/utility.py:186]: How about [os.PathLike](https://docs.python.org/3/library/os.html#os.PathLike)
File descriptors should technically also work with this function but fast64 never uses them so meh

File: fast64_internal/utility.py [hunk 2]
```diff
@@ -137,33 +183,56 @@ def restoreHiddenState(hiddenState):
...
+    """Read a file and return its contents"""
```
Comment (Lilaa3) [inline fast64_internal/utility.py:187]: Specify that its expecting a text file with utf-8 encoding

**Review by Lilaa3 — CHANGES_REQUESTED** [review]
I will review more later

**Pull request closed without merging**